src/main/java/com/example/userrestapi/controller/UserController.java

```diff
@@ -21,30 +23,16 @@
                                                       @RequestParam(value = "surname", required = false) String surname,
                                                       @RequestParam(value = "email", required = false) String email) {
 
-        List<User> users = new ArrayList<User>();
-
-        if (name != null) {
-            users = repository.findByName(name);
-            return new ResponseEntity<>(users, HttpStatus.OK);
-        }
-        if (surname != null) {
-            users = repository.findBySurname(surname);
-            return new ResponseEntity<>(users, HttpStatus.OK);
-        }
-        if (email != null) {
-            users = repository.findByEmail(email);
-            return new ResponseEntity<>(users, HttpStatus.OK);
-        }
+        List<User> users = repository.findByParams(name, surname, email);
 
-        users = repository.findAll();
         return new ResponseEntity<>(users, HttpStatus.OK);
     }
 
     @RequestMapping(value = "/{id}", method = RequestMethod.GET)
     public ResponseEntity<User> findUserById(@PathVariable Long id) {
         User user = repository.findById(id);
 
-        if (repository.containsUserWithId(id)) {
+        if (repository.findById(id) != null) {
```

Comment:
don't call repository.findById again, you already have this data in "user" variable
```suggestion
        if (user != null) {
```